src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Lower.java

```diff
@@ -2562,6 +2600,61 @@ private void visitEnumConstantDef(JCVariableDecl var, int ordinal) {
         varDef.args = varDef.args.
             prepend(makeLit(syms.intType, ordinal)).
             prepend(makeLit(syms.stringType, var.name.toString()));
+        if (!useDynamicConstants || !varDef.args.stream().allMatch(a -> foldableEnumConstantArg(a))) {
+            // TODO - synthesize a factory method and use condy
+            return;
+        }
+        if (varDef.def != null) {
+            // TODO - handle initializers that declare anonymous classes
+            return;
+        }
+        if (var.type.tsym.enclClass().packge().modle == syms.java_base) {
+            // avoid using MethodHandle in enums referenced from MethodHandle's clinit
+            return;
+        }
+        ListBuffer<LoadableConstant> params = new ListBuffer<>();
+        for (JCExpression arg : varDef.args) {
+            params.add(foldEnumConstantArg((JCLiteral) arg));
+        }
+
+        List<Type> bsm_staticArgs = List.of(syms.methodHandleLookupType,
+                syms.stringType,
+                new ClassType(syms.classType.getEnclosingType(),
+                        List.of(syms.botType),
+                        syms.classType.tsym),
+                syms.methodHandleType,
+                types.makeArrayType(syms.objectType));
+        MethodSymbol bsm = rs.resolveInternalMethod(var.pos(), attrEnv, syms.constantBootstrapsType,
+                names.invoke, bsm_staticArgs, List.nil());
+        MethodHandleSymbol toCall = ((MethodSymbol) varDef.constructor).asHandle();
+        params.prepend(toCall);
+        DynamicVarSymbol dynSym = new DynamicVarSymbol(bsm.name, bsm.owner, bsm.asHandle(),
+                var.type, params.toArray(new LoadableConstant[0]));
```

Comment:
The **condy** name of the dynamic constants should probably be the same as the enum constant name:
```suggestion
        DynamicVarSymbol dynSym = new DynamicVarSymbol(var.name, bsm.owner, bsm.asHandle(),
                var.type, params.toArray(new LoadableConstant[0]));
```

So that instead of:
```
BootstrapMethods:
  0: #81 REF_invokeStatic E.$values:(Ljava/lang/invoke/MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/Class;[LE;)[LE;
    Method arguments:
      #44 #1:invoke:LE;
      #51 #2:invoke:LE;
      #55 #3:invoke:LE;
```

it ends up being:
```
BootstrapMethods:
  0: #81 REF_invokeStatic E.$values:(Ljava/lang/invoke/MethodHandles$Lookup;Ljava/lang/String;Ljava/lang/Class;[LE;)[LE;
    Method arguments:
      #44 #1:ONE:LE;
      #51 #2:TWO:LE;
      #55 #3:THREE:LE;
```

Reply:
Done, thanks!